### DISCUSSION TITLE
Race condition during table insertion

### DISCUSSION THREAD

**Discussion by sundancecloud** [stevengharris/MarkupEditor#141] [General]
Hi,

On Mac Catalyst, when dragging to insert a table:

After letting the mouse button up, the "onEnded" handler is invoked and the rows&cols are picked up for table creation. This is all good.

However, because the mouse for a brief moment is still over the popup, the "onHover" handler is  also invoked. This promptly sets rows&cols to 0.

The race condition is that onEnded may or may not be serviced before onHover. If onHover is invoked first the rows&cols are set to zero .... and then onEnded is serviced, but then no table of 0x0 will be created.

A proposed fix, is to modify the onHover() handler:

```
// Don't clear if a drag has just happened.
if !dragged {
       rows = 0
       cols = 0
 }
```

On a side note, I would like to thank Steven for this great piece of software!

**Comment by stevengharris** [comment]
Thank you for raising the issue with all these details. I'm moving this to the Issues area and will try to reproduce and fold-in the fix soon. And thanks for the kind words :-).

**Comment by stevengharris** [comment]
Closed with https://github.com/stevengharris/MarkupEditor/issues/142
